### ISSUE TITLE
Expand the library of functions for Gaffer types

### ISSUE THREAD

**Issue by p013570** [gchq/Gaffer#2211]
Functions such as:
- TypeSubTypeValueToTuple (converts TypeSubTypeValue into a String Tuple with 'type', 'subType', 'value' as keys. A test for this would be to convert a TypeSubTypeValue into a Tuple, then extract the Value and call Length then ToString. Then extract the Type and call SetValue('Length')
- TypeValueToTuple (same as above but for TypeValue objects)
- ToTrailingWildcardPair (produces a Pair of EntityIds to allow a follow on query using the GetElementsInRanges operation. Implements KorypheFunction<Object, Pair<EntityId, EntityId>>. Allows the user to provide the end of the range (defaults to `~`). For example, takes the String 'value1' and the end of the range `'~'` and produces `new Pair(new EntitySeed('value1'), new EntitySeed('value1~'))`. 
- ReduceRelatedElements (takes an iterable of elements and reduces elements that have related vertices. For example A1-relates-to->A2,  B1-relates-to->B2.  A1->B1 and A2->B2 should be merged into 1 edge A1->B1 with properties on the edge relatedSourceVertices: ['A2'] and relatedDestinationVertices: ['B2'].

**Comment by p013570** [comment]
I will be working on and off for the next couple of months.

**Comment by p013570** [comment]
Working on branch gh-2211-functions

**Comment by GCHQDev404** [comment]
blocked by gh-2443
